### PR TITLE
[#361] Put all warnings under `if`s to always have them all

### LINKED ISSUE #361: Put all warnings under `if`s to always have them all
Blocked by #246 

Once we put all warnings into common stanzas, we can always put all GHC options and warnings under `if` conditionals. This is more useful because we can automatically see all warnings and always have them even when the project is updated.

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #361

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ The changelog is available [on GitHub][2].
 * [#363](https://github.com/kowainik/summoner/issues/363):
   Move from `generic-deriving` to `generic-data`.
   (by [@chshersh](https://github.com/chshersh))
+* [#361](https://github.com/kowainik/summoner/issues/361):
+  Always put all warnings in `ghc-options` inside common stanza under `if`s.
```
Comment (vrom911) [inline CHANGELOG.md]: Let's add an example of `if` in here if you don't mind :slightly_smiling_face: 
Also, maybe makes sense to mention that the condition is on GHC in these `if`s? Or only one of these two options is fine as well, what do you think?

File: summoner-cli/src/Summoner/License.hs [hunk 2]
```diff
@@ -65,11 +64,6 @@ newtype License = License
 instance FromJSON License where
     parseJSON = withObject "License" $ \o -> License <$> o .: "body"
 
--- | As it will be shown in the @cabal@ file.
-cabalLicense :: LicenseName -> Text
-cabalLicense None = "AllRightsReserved"
-cabalLicense l    = show l
-
```
Comment (vrom911) [inline summoner-cli/src/Summoner/License.hs]: That is a nice simplification!

File: summoner-cli/src/Summoner/Template/Cabal.hs [hunk 3]
```diff
@@ -95,15 +94,42 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
...
     |] <> defaultExtensions
 
     baseBounds :: Text
     baseBounds = cabalBaseVersions settingsTestedVersions
 
+    ghcOptions :: Text
+    ghcOptions = case settingsGhcOptions of
+        [] -> defaultGhcOptions
+        x:xs ->
+            let customGhcOptions = T.intercalate "\n" $ x : map (T.replicate 21 " " <>) xs in
```
Comment (vrom911) [inline summoner-cli/src/Summoner/Template/Cabal.hs]: Could `intercalateMap` function be handy in here?

Comment (chshersh) [reply summoner-cli/src/Summoner/Template/Cabal.hs]: Yes, it can help here indeed! I missed that

Comment (vrom911) [reply summoner-cli/src/Summoner/Template/Cabal.hs]: Or not.. looking into it I see, that the logic here is a bit different. Sorry, false alarm!

**Pull request merged**